cylc/flow/network/schema.py

```diff
@@ -309,27 +322,57 @@ class SortArgs(InputObjectType):
 
 
 async def get_workflows(root, info, **args):
+    """Get filtered workflows."""
+    # Set subscription info
+    if 'sub_id' in info.context:
+        args['sub_id'] = info.context['sub_id']
+        if root and field_name in root:
+            args['ids'] = [root[field_name].id]
+
     args['workflows'] = [parse_workflow_id(w_id) for w_id in args['ids']]
     args['exworkflows'] = [parse_workflow_id(w_id) for w_id in args['exids']]
     resolvers = info.context.get('resolvers')
     return await resolvers.get_workflows(args)
 
 
+async def get_workflow_by_id(root, info, **args):
+    """Return single workflow element."""
+    if 'sub_id' in info.context:
+        args['sub_id'] = info.context['sub_id']
+        field_name = to_snake_case(info.field_name)
+        if root and field_name in root:
+            args['id'] = root[field_name].id
+
+    args['workflow'] = args['id']
+    resolvers = info.context.get('resolvers')
+    return await resolvers.get_workflow_by_id(args)
+
+
 async def get_nodes_all(root, info, **args):
     """Resolver for returning job, task, family nodes"""
     field_name = to_snake_case(info.field_name)
-    field_ids = getattr(root, field_name, None)
+    if isinstance(root, dict):
+        field_ids = root.get(field_name, None)
+    else:
+        field_ids = getattr(root, field_name, None)
```

Comment:
Shall I put a link to Graphene documentation?

Reply:
What's worse; a little duplication or more function calls?

Reply:
Due to the number of duplicates (of this block and others) I think it would be better to centralise, the chances of one block not getting updated when a change is needed or a block not getting added in the right places is very high (reviewers miss things).